crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -548,17 +588,16 @@ impl<'src> Parser<'src> {
         let slice_start = self.node_start();
         let mut slice = self.parse_slice();
 
+        // If there are more than one element in the slice, we need to create a tuple
+        // expression to represent it.
         if self.eat(TokenKind::Comma) {
             let mut slices = vec![slice];
-            #[allow(deprecated)]
-            self.parse_separated(
+
+            slices.extend_from_slice(&self.parse_comma_separated_list_into_vec(
+                RecoveryContextKind::Slices,
+                Parser::parse_slice,
                 true,
-                TokenKind::Comma,
-                TokenSet::new([TokenKind::Rsqb]),
-                |parser| {
-                    slices.push(parser.parse_slice());
-                },
-            );
+            ));
```

Comment:
I recommend using `parse_comma_separated_list` here because I don't know if the compiler recognize that it can directly move the elements into `slices` without cloning the expressions (which is what `extend_from_slice`).